File: src/main/java/com/code/Product_management/Service/productservice.java
package com.code.Product_management.Service;

import com.code.Product_management.Model.Product;

import java.util.List;

public interface productservice {
    public Product saveProduct (Product product);
    public List<Product> getAllProduct();
//    public List<Product> findByproductName(String productName);
    public Product getProductById(String id);
    public String deleteProduct(String id);
}

File: product_ui/src/Service/Product.service.js
import axios from "axios";

const API_URL = "http://localhost:8089";

class ProductService {
  saveProduct(product) {
    return axios.post(API_URL + "/saveProduct", product);
  }

  getAllProduct() {
    return axios.get(API_URL + "/product");
  }

  getProductById(id) {
    return axios.get(API_URL + "/product" + id);
  }

  deleteProduct(id) {
    return axios.get(API_URL + "/deleteProduct" + id);
  }

  editProduct(product) {
    return axios.post(API_URL + product.id, product);
  }
}

export default new ProductService();
